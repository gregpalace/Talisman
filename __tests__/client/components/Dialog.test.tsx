import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Dialog from '../../../src/client/components/Dialog';

describe('React Unit Tests', () => {
  let wrapper: any;

  const props = {
    goBack: true,
    onKeyPress: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Dialog {...props} />);
  });

  describe('<Shop />', () => {
    xit('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a div with a back button', () => {
      expect(wrapper.find('div')).toBeDefined();
      expect(wrapper.find('button').prop('className')).toEqual(
        'dialog__back-button'
      );
    });

    it('should exit out of the current dialog window', () => {
      let button = wrapper.find('button');
      expect(button).toBeInstanceOf(Function);
    });
  });
});
