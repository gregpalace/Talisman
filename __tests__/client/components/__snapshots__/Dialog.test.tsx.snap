// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Unit Tests <Shop /> should render correctly 1`] = `
<div
  className="dialog__container white-border"
>
  <button
    className="dialog__back-button"
    onClick={[MockFunction]}
  >
    <i
      className="fa fa-arrow-left"
    />
  </button>
</div>
`;
